Cesium3DTilesSelection/src/WebMapServiceRasterOverlay.cpp

```diff
@@ -0,0 +1,410 @@
+          [pOwner,
+           asyncSystem,
+           pAssetAccessor,
+           credit,
+           pPrepareRendererResources,
+           pLogger,
+           options = this->_options,
+           url = this->_baseUrl,
+           headers = this->_headers,
+           reportError](const std::shared_ptr<IAssetRequest>& pRequest)
+              -> std::unique_ptr<RasterOverlayTileProvider> {
+            const IAssetResponse* pResponse = pRequest->response();
+            if (!pResponse) {
+              reportError(
+                  pRequest,
+                  "No response received from web map service.");
+              return nullptr;
+            }
+
+            const gsl::span<const std::byte> data = pResponse->data();
+
+            tinyxml2::XMLDocument doc;
+            const tinyxml2::XMLError error = doc.Parse(
+                reinterpret_cast<const char*>(data.data()),
+                data.size_bytes());
+            if (error != tinyxml2::XMLError::XML_SUCCESS) {
+              reportError(pRequest, "Could not parse web map service XML.");
+              return nullptr;
+            }
+
+            tinyxml2::XMLElement* pRoot = doc.RootElement();
+            if (!pRoot) {
+              reportError(
+                  pRequest,
+                  "Web map service XML document does not have a root "
+                  "element.");
+              return nullptr;
+            }
+
+            std::string validationError;
+            if (!validateCapabilities(pRoot, options, validationError)) { 
+              reportError(pRequest, validationError);
+              return nullptr;
+            }
+
+            const auto projection = CesiumGeospatial::GeographicProjection();
+
+            CesiumGeospatial::GlobeRectangle tilingSchemeRectangle =
+                CesiumGeospatial::GeographicProjection::MAXIMUM_GLOBE_RECTANGLE;
+
+            CesiumGeometry::Rectangle coverageRectangle =
+                projectRectangleSimple(projection, tilingSchemeRectangle);
+
+            const int rootTilesX = 2;
+            const int rootTilesY = 1;
+            CesiumGeometry::QuadtreeTilingScheme tilingScheme(coverageRectangle, rootTilesX, rootTilesY);
+
+            return std::make_unique<WebMapServiceTileProvider>(
+                *pOwner,
+                asyncSystem,
+                pAssetAccessor,
+                credit,
+                pPrepareRendererResources,
+                pLogger,
+                projection,
+                tilingScheme,
+                coverageRectangle, 
+                url,
+                headers,
+                options.version.value_or("1.3.0"),
+                options.layers,
+                options.fileExtension.has_value() ? "." + options.fileExtension.value() : "",
+                options.tileWidth.value_or(256),
+                options.tileHeight.value_or(256),
+                options.minimumLevel.value_or(1),
+                options.maximumLevel.value_or(14)
```

Comment:
I think it would be better if these default values were set in `WebMapServiceRasterOverlayOptions`, rather than being optional there and quietly given values here. It would make the defaults more obvious.